### PR TITLE
docs: split plugin file into chapter files

### PR DESCRIPTION
**Pull request by DjamilaMamedov**
The Plugin.rst file has been split into the chapterfiles: DirectorPlugins.rst.inc, FileDaemonPlugins.rst.inc and StorageDaemonPlugins.rst.inc .
The chapterfiles had been split into one file per plugin.
### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)

##### General

- [x] PR name is meaningful
- [x] Purpose of the PR is understood
- [x] Separate commit for this PR in the CHANGELOG.md, PR number referenced is same
- [x] Commit descriptions are understandable and well formatted

##### Source code quality

- ~~[ ] Source code changes are understandable~~
- ~~[ ] Variable and function names are meaningful~~
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR
- [x] `bareos-check-sources --since-merge` does not report any problems
- [x] `git status` should not report modifications in the source tree after building and testing

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,6 +25,8 @@ and since Bareos version 20 this project adheres to [Semantic Versioning](https:
 ### Security
 
 ### Documentation
+-split plugin file into chapter files, removed wrong includes, created a file for each Plugin[PR #1046]
```
Comment (joergsteffens) [inline CHANGELOG.md:28]:
```suggestion
- split the very long `Plugins.rst` file into one file per Bareos plugin [PR #1046]
```

File: docs/manuals/source/TasksAndConcepts/Plugins.rst [hunk 2]
```diff
@@ -313,2251 +313,10 @@ with the **python3** plugin (which uses Python 3).
 
    It is not possible to use the python plugin to restore backups created with
    the python3 plugin. Once switched, you need to stay on python3.
+
```
Comment (bruno-at-bareos) [inline docs/manuals/source/TasksAndConcepts/Plugins.rst:316]:
```suggestion

.. include:: Plugins/DirectorPlugins.rst.inc
.. include:: Plugins/FileDaemonPlugins.rst.inc 
.. include:: Plugins/StorageDaemonPlugins.rst.inc
```
Better to keep Director first, as it is a logical follow up of Python.

File: docs/manuals/source/TasksAndConcepts/Plugins/FileDaemonPlugins.rst.inc [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+   
+   
+
+.. include:: Plugins/FileDaemonPlugins/ApacheLibcloudPlugin.rst.inc 
+.. include:: Plugins/FileDaemonPlugins/BpipePlugin.rst.inc 
+.. include:: Plugins/FileDaemonPlugins/CephfsPlugin.rst.inc 
+.. include:: Plugins/FileDaemonPlugins/GlusterFsPlugin.rst.inc 
+.. include:: Plugins/FileDaemonPlugins/LdapPlugin.rst.inc 
+.. include:: Plugins/FileDaemonPlugins/MariaDBmariabackupPlugin.rst.inc
+.. include:: Plugins/FileDaemonPlugins/MariaDBPlugin.rst.inc
```
Comment (bruno-at-bareos) [inline docs/manuals/source/TasksAndConcepts/Plugins/FileDaemonPlugins.rst.inc:21]:
```suggestion

```

Comment (bruno-at-bareos) [inline docs/manuals/source/TasksAndConcepts/Plugins/FileDaemonPlugins.rst.inc:21]:
```suggestion

```
There's nothing useful inside it, Corresponding file should be removed

File: docs/manuals/source/TasksAndConcepts/Plugins/FileDaemonPlugins/MariaDBPlugin.rst.inc [hunk 4]
```diff
@@ -0,0 +1,4 @@
+MariaDB Plugin
```
Comment (bruno-at-bareos) [inline docs/manuals/source/TasksAndConcepts/Plugins/FileDaemonPlugins/MariaDBPlugin.rst.inc:1]: This file need to be removed, nothing useful in.

File: docs/manuals/source/TasksAndConcepts/Plugins/StorageDaemonPlugins/AutoxflateSd.rst.inc [hunk 5]
```diff
@@ -0,0 +1,60 @@
+.. _plugin-autoxflate-sd:
+
+autoxflate-sd
+~~~~~~~~~~~~~
+
+:index:`\ <single: Plugin; autoxflate-sd>`\
```
Comment (bruno-at-bareos) [inline docs/manuals/source/TasksAndConcepts/Plugins/StorageDaemonPlugins/AutoxflateSd.rst.inc:6]:
```suggestion
.. index::
   single: Plugin; autoxflate-sd
```

File: docs/manuals/source/TasksAndConcepts/Plugins/StorageDaemonPlugins/PythonSdPlugin.rst.inc [hunk 6]
```diff
@@ -0,0 +1,7 @@
+python-sd Plugin
+~~~~~~~~~~~~~~~~
+
+:index:`\ <single: Plugin; Python; Storage Daemon>`\
```
Comment (bruno-at-bareos) [inline docs/manuals/source/TasksAndConcepts/Plugins/StorageDaemonPlugins/PythonSdPlugin.rst.inc:4]:
```suggestion
.. index::
   single: Plugin; Python; Storage Daemon
```
Use sphinx syntax for index

File: docs/manuals/source/TasksAndConcepts/Plugins/StorageDaemonPlugins/ScsicryptoSd.rst.inc [hunk 7]
```diff
@@ -0,0 +1,197 @@
+scsicrypto-sd
+~~~~~~~~~~~~~
+
+:index:`\ <single: Plugin; scsicrypto-sd>`\
```
Comment (bruno-at-bareos) [inline docs/manuals/source/TasksAndConcepts/Plugins/StorageDaemonPlugins/ScsicryptoSd.rst.inc:4]:
```suggestion
.. index::
   single: Plugin; scsicrypto-sd
```

File: docs/manuals/source/TasksAndConcepts/Plugins/StorageDaemonPlugins/ScsitapealertSd.rst.inc [hunk 8]
```diff
@@ -0,0 +1,7 @@
+scsitapealert-sd
+~~~~~~~~~~~~~~~~
+
+:index:`\ <single: Plugin; scsitapealert-sd>`\
```
Comment (bruno-at-bareos) [inline docs/manuals/source/TasksAndConcepts/Plugins/StorageDaemonPlugins/ScsitapealertSd.rst.inc:4]:
```suggestion
.. index::
   single: Plugin; scsitapealert-sd
```

File: docs/manuals/source/TasksAndConcepts/Plugins/DirectorPlugins/PythonDirPlugin.rst.inc [hunk 9]
```diff
@@ -0,0 +1,100 @@
+.. _director-python-plugin:
+
+python-dir Plugin
+~~~~~~~~~~~~~~~~~
+
+:index:`\ <single: Plugin; Python; Director>`\
```
Comment (bruno-at-bareos) [inline docs/manuals/source/TasksAndConcepts/Plugins/DirectorPlugins/PythonDirPlugin.rst.inc:6]:
```suggestion
.. index::
   single: Plugin; Python; Director
```

**Review by bruno-at-bareos — CHANGES_REQUESTED** [review]
I've made some suggestion to also fix all index syntax to use the sphinx one.
I think that mariadbplugin.rst can be remove as it serve nothing useful.

**Review by bruno-at-bareos — CHANGES_REQUESTED** [review]
There's still some needed changes, as I found some mispelled lower/upper case name in plugin file name.
The following files need to be renamed.
Problem with lower/upper case name of split plugin files

- docs/manuals/source/TasksAndConcepts/Plugins/FileDaemonPlugins/GlusterFsPlugin.rst.inc -> GlusterFSPlugin.rst.inc
- docs/manuals/source/TasksAndConcepts/Plugins/FileDaemonPlugins/LdapPlugin.rst.inc -> LDAPPlugin.rst.inc
- docs/manuals/source/TasksAndConcepts/Plugins/FileDaemonPlugins/MsSqlPlugin.rst.inc -> MSSqlPlugin.rst.inc
- docs/manuals/source/TasksAndConcepts/Plugins/FileDaemonPlugins/MySqlPlugin.rst.inc -> MySQL.rst.inc
- docs/manuals/source/TasksAndConcepts/Plugins/FileDaemonPlugins/PostgreSqlPlugin.rst.inc -> PostgreSQLPlugin.rst.inc

**Review by bruno-at-bareos — APPROVED** [review]
The build doc fail actually and that need to be fixed.

**Review by joergsteffens — APPROVED** [review]

**Pull request merged**